xlators/protocol/client/src/client.c

```diff
@@ -2662,12 +2662,12 @@ init(xlator_t *this)
 
     ret = client_init_rpc(this);
 out:
+    if (ret == -1 && conf) {
+        GF_FREE(conf);
+    }
```

Comment:
No, even this is not correct.
Actually this->fini(this) is responsible for freeing conf correctly after freeing any memory allocated to any field of conf. Check the function fini just below this function definition.

The real problem is tricky to find out and you have to go through the definition of function and the steps given in this coverity ID.
Problem is that this->fini(this) can free the conf only if "this->private" is having pointer to conf which is assigned on line number 2638. However, you can come to "out" after allocating memory to conf but before assigning conf to this->private. From line no 2628. Check definition of GF_OPTION_INIT
.
It should be like this  - 
if (ret == -1) {
      if (this->private == conf) {
           this->fini(this);
      } else {
           GF_FREE(conf);
      }
    }